lib/sim.c

```diff
@@ -1181,7 +1247,102 @@ sim_simulate(sim_t *self, uint64_t max_events)
     return ret;
 }
 
+int
+sim_simulate_arg(sim_t *self) 
+{
+    int ret = 0;
+    double recomb_rate, coal_rate, total_rate; 
+    individual_t **S = self->intersected_buffer;
+    unsigned int S_size = 0;
+    avl_node_t *node;
+    uintptr_t int_ptr;
+    uint64_t id;
+    individual_t *ind;
+    /* An arbitrary radius and parental location for using existing spatial 
+     * functions to complete events */
+    double z[] = {self->pixel_size / 2.0, self->pixel_size / 2.0}, r = self->pixel_size / 4.0;
```

Comment:
I don't know about c99, but to me this looks identical to declarations and initialisations of the analogous z[] variables at the top of sim_simulate and sim_non_reproduction_event. Could you be more specific about what it should look like instead? Just having z[] and r declared on separate lines?